lib/DDA/ContextDDA.cpp

```diff
@@ -315,7 +315,15 @@ bool ContextDDA::isHeapCondMemObj(const CxtVar& var, const StoreSVFGNode*)
     assert(mem && "memory object is null??");
     if(mem->isHeap())
     {
-        if(const Instruction* mallocSite = SVFUtil::dyn_cast<Instruction>(mem->getRefVal()))
+        if (!mem->getRefVal()) {
+            PAGNode *pnode = _pag->getPAGNode(getPtrNodeID(var));
+            assert((pnode->getNodeKind() == PAGNode::DummyObjNode || 
+                pnode->getNodeKind() == PAGNode::DummyValNode ||
+                pnode->getNodeKind() == PAGNode::CloneDummyObjNode) &&
+                "emtpy refVal in non-dummy object");
+            return false;
```

Comment:
This should return true since we assume dummy each heap object is not a singleton.